### PR TITLE
Remove unprocessedRegions usage

### PR DESCRIPTION
**Pull request by jcmorrow**
The whole `ensureProcessed` command is probably going to either get deleted or change considerably in the near future. It seems like basically all of this is not really powering any important functionality at the moment. The SourceMap visualizer has a comment about showing a loading bar, but actually removing this has no affect on the source map viewer loading screen.

Blocks https://linear.app/replay/issue/BAC-2871/make-sure-that-the-dispatcher-waiting-on-regions-is-not-a-problem

### CODE REVIEW

File: packages/protocol/thread/thread.ts [hunk 1]
```diff
@@ -455,23 +435,23 @@ class _ThreadFront {
...
-          this.sessionId!,
-          pauseId
-        );
+        {
+          expression: text,
+          pure: abilities.supportsPureEvaluation && pure,
+        },
+        this.sessionId!,
+        pauseId
+      );
```
Comment (jcmorrow) [inline packages/protocol/thread/thread.ts:454]: My `prettier` made these indent changes. Is it wrong or should I leave them in?

File: pages/recording/[id]/sourcemap/[sourceId].tsx [hunk 2]
```diff
@@ -54,14 +54,6 @@ async function loadSourceMap(
     }
     const { sessionId } = await client.Recording.createSession({ recordingId });
 
-    // this will show a progress bar in the LoadingScreen, note that the
-    // sourcemap visualizer is shown as soon as the requested source and its sourcemap
-    // have been loaded, which may happen before the progress bar reaches 100%
-    client.Session.ensureProcessed({ level: "basic" }, sessionId);
-    client.Session.addUnprocessedRegionsListener(regions =>
-      store.dispatch(onUnprocessedRegions(regions))
-    );
```
Comment (jcmorrow) [inline pages/recording/[id]/sourcemap/[sourceId].tsx:63]: This was not actually showing a progress bar.

File: src/ui/reducers/app.ts [hunk 3]
```diff
@@ -64,13 +64,6 @@ const appSlice = createSlice({
     setMouseTargetsLoading(state, action: PayloadAction<boolean>) {
       state.mouseTargetsLoading = action.payload;
     },
-    durationSeen(state, action: PayloadAction<number>) {
-      // If this is the longest duration we have seen yet, we should update the
-      // store
-      if (action.payload > state.recordingDuration) {
-        state.recordingDuration = action.payload;
-      }
-    },
```
Comment (jcmorrow) [inline src/ui/reducers/app.ts:73]: We should be using `getEndpoint` for this. I have not seen anything not work with this deleted, but lmk if there's something else I should check

File: packages/protocol/thread/thread.ts (hunk 1 above)
Comment (markerikson) [reply packages/protocol/thread/thread.ts:454]: ~~Eh, if it passes lint let's leave it.~~

well lint failed :)  double-check settings I guess?

File: src/ui/reducers/app.ts (hunk 3 above)
Comment (markerikson) [reply src/ui/reducers/app.ts:73]: I do see several references to `recordingDuration`, particularly around playback.  Looks like it also gets passed via context into some of the `replay-next` functionality.  Can we swap this out with the better API right now?

Comment (jcmorrow) [reply src/ui/reducers/app.ts:73]: Yeah I'll see what I can do in a bit!

Comment (jcmorrow) [reply src/ui/reducers/app.ts:73]: It seems like [this line](https://github.com/replayio/devtools/blob/main/src/ui/actions/timeline.ts#L115) is doing exactly what I was suggesting?

File: packages/protocol/thread/thread.ts (hunk 1 above)
Comment (jcmorrow) [reply packages/protocol/thread/thread.ts:454]: Ah, shoot. Well, I'm trying out lazy.vim and it has some different lsp configs than I usually use 🤔

File: src/ui/reducers/app.ts (hunk 3 above)
Comment (markerikson) [reply src/ui/reducers/app.ts:73]: Hmm. Seems like, it, yeah?  (Also, putting on my Redux hat: I _hate_ that we have an action named `setTimelineState`. That is literally the exact opposite mentality of how we want people to _think_ about using Redux :(  )

Comment (jcmorrow) [reply src/ui/reducers/app.ts:73]: Fair!

**Comment by jcmorrow** [comment]
I'm not sure why the 🦄 would have changed with this PR? @markerikson is that a thing I should be concerned about?

File: src/ui/actions/session.ts [hunk 4]
```diff
@@ -320,7 +320,6 @@ export function createSocket(
       // We don't want to show the non-dev version of the app for node replays.
       if (recordingTarget === "node") {
         dispatch(setViewMode("dev"));
-        dispatch(onLoadingFinished());
```
Comment (markerikson) [inline src/ui/actions/session.ts:323]: How did we end up dispatching this _twice_ in the case where it's a Node recording?  Looks like.... Jason added that in June 2022.

**Review by markerikson — APPROVED** [review]

File: src/ui/actions/session.ts (hunk 4 above)
Comment (jcmorrow) [reply src/ui/actions/session.ts:323]: Haha, yeah I dunno, many hands make light work... but also sometimes the work is light and also a little confused!

**Pull request merged**